gratipay/utils/team_history.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from datetime import datetime
+from decimal import Decimal as D
+
+
+def get_end_of_year_totals(db, team, year):
+
+    received = db.one("""
+        SELECT COALESCE(sum(amount), 0) AS Received
+          FROM payments
+         WHERE team = %(team)s
+           AND extract(year from timestamp) = %(year)s
+           AND amount > 0
+           AND direction='to-team';
+    """, locals() )
+
+    if received is None:
+       received = D(0.00)
+
+    distributed = db.one("""
+        SELECT COALESCE(sum(amount), 0) AS Distributed
+          FROM payments
+         WHERE team = %(team)s
+           AND extract(year from timestamp) = %(year)s
+           AND amount > 0
+           AND direction='to-participant';
+    """, locals())
+
+    if distributed is None:
+        distributed = D(0.00)
+
+    return received, distributed
+
+
+def iter_team_payday_events(db, team, year=None):
+    """Yields payday events for the given Team.
+    """
+    current_year = datetime.utcnow().year
+    year = year or current_year
+
+    payments = db.all("""
+        SELECT payments.*, paydays.ts_start
+          FROM payments, paydays
+         WHERE payments.payday = paydays.id
```

Comment:
This looks like a use case for `JOIN` query. 

ie. Instead of 

```
SELECT payments.*, paydays.ts_start
FROM payments, paydays
WHERE payments.payday = paydays.id
```
use a single `JOIN`: 

```
SELECT payments.*, paydays.ts_start
FROM payments JOIN paydays ON payments.payday = paydays.id
```

I find that easier to read (the intent of having multiple tables + their relationship is clear), and [from what I've read](http://stackoverflow.com/questions/377729/performance-single-join-select-vs-multiple-simple-selects) - databases can do more intelligent optimizations with `JOIN`s compared to multiple `SELECT`s.